python/lib/dependabot/python/update_checker/index_finder.rb

```diff
@@ -120,6 +120,8 @@ def pyproject_index_urls
           sources.each do |source|
             if source["default"]
               urls[:main] = source["url"]
+            elsif source["name"] == "PyPI"
```

Comment:
Perhaps check for case-insensitive match (is that already the case?)?

Reply:
I just chose the string that gets added by running `poetry source add pypi` (the command that the poetry warning suggests). To me that sounds fine, but I don't feel strongly about it 🤷🏻‍♂️

Reply:
Double-checked on this, trying some workarounds locally.

Poetry will check itself for the source name matching `pypi`, case insensitive: both `PyPI` and `pypi` work from that primary source.

Attempting to add a `url = "https://pypi.org/simple"` field to the TOML for either of those names throws an error from Poetry: `The PyPI repository cannot be configured with a custom url.`

Therefore, a case-insensitive match would be safest. I'm not too familiar with Ruby, but my quick search turns up `casecmp`, so maybe this would help?

```suggestion
            elsif source["name"].casecmp?("PyPI")
```